### PR TITLE
Headers

### PR DESCRIPTION
**Pull request by moozzyk**
Adding support for HTTP headers.

### CODE REVIEW

File: src/signalrclient/request_sender.h [hunk 1]
```diff
@@ -13,10 +13,13 @@ namespace signalr
...
+            const utility::string_t& connection_data, const utility::string_t& query_string,
+            std::unordered_map<utility::string_t, utility::string_t> headers);
         pplx::task<void> start(web_request_factory& request_factory, const web::uri& base_url, transport_type transport,
-            const utility::string_t& connection_token, const utility::string_t& connection_data, const utility::string_t& query_string);
+            const utility::string_t& connection_token, const utility::string_t& connection_data, const utility::string_t& query_string,
+            std::unordered_map<utility::string_t, utility::string_t> headers);
         pplx::task<utility::string_t> abort(web_request_factory& request_factory, const web::uri& base_url, transport_type transport,
-            const utility::string_t& connection_token, const utility::string_t& connection_data, const utility::string_t& query_string);
+            const utility::string_t& connection_token, const utility::string_t& connection_data, const utility::string_t& query_string,
+            std::unordered_map<utility::string_t, utility::string_t> headers);
```
Comment (BrennanConroy) [inline src/signalrclient/request_sender.h]: These 3 functions take the unordered_map by value

Comment (moozzyk) [inline src/signalrclient/request_sender.h]: I thought I fixed this. Will double check and fix.

**Comment by BrennanConroy** [comment]
Other than comment, looks good :sheep: :it:

**Comment by moozzyk** [comment]
Thanks @BrennanConroy and @halter73 !

**Pull request closed without merging**